backend/routes.py

```
@@ -35,6 +37,63 @@ def register_event_handlers():
 
+@router.post("/repo-stats")
+async def get_repo_stats(request: RepoRequest):
+    """
+    Get repository statistics from GitHub
+    """
+    try:
+        # Parse repo URL to extract owner and repo name
+        repo_url = request.repo_url.rstrip('/')
+        parts = repo_url.split('github.com/')[-1].split('/')
+        if len(parts) < 2:
+            raise HTTPException(status_code=400, detail="Invalid GitHub repository URL")
```

Ruff (0.14.11)

Abstract `raise` to an inner function

(TRY301)